src/main/java/edu/escuelaing/ieti/matchmaking/services/impl/ActivityServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package edu.escuelaing.ieti.matchmaking.services.impl;
+
+import edu.escuelaing.ieti.matchmaking.exception.MatchmakingException;
+import edu.escuelaing.ieti.matchmaking.persistence.ActivityRepository;
+import edu.escuelaing.ieti.matchmaking.services.ActivityService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import edu.escuelaing.ieti.matchmaking.model.Activity;
+
+import java.util.List;
+
+@Service
+public class ActivityServiceImpl implements ActivityService {
+
+    @Autowired
+    private ActivityRepository inMemoryRepository;
+
+    @Override
+    public Activity create(Activity activity) throws MatchmakingException {
```

Comment:
So, this exception gets propagates towards the controller and then what? Will the controller be in charge of catching the exception and handling the error? Take a look into the @ControllerAdvice and @ExceptionHandler annotations.